### DISCUSSION TITLE
How to write a keybind to open current file in mini.files and how to use show_at_cursor() on git diff command to go to buffer which is already open.

### DISCUSSION THREAD

**Discussion by 231tr0n** [nvim-mini/mini.nvim#1209] [Q&A] [answered]
Hi @echasnovski! I had two issues when using mini.files and mini-git in my workflow.
1. How do I open current file location in mini.files instead of opening pwd and navigating to it?
2. When using MiniGit.show_at_cursor() on ``:Git diff`` commands diffs, it opens up a new window instead of navigating to the existing one. Also the new window does not have an Lspattached to it and hence I will have to manually open the file again. How do you think I can solve this issue.

**Comment by echasnovski** [comment, marked as answer]
> 1. How do I open current file location in mini.files instead of opening pwd and navigating to it?

It is [in documentation of the main `open()` function](https://github.com/echasnovski/mini.nvim/blob/8413efde2546be4dec9ea424bc6af346a14dffbe/doc/mini-files.txt#L694-L708).

> 2. When using MiniGit.show_at_cursor() on `:Git diff` commands diffs, it opens up a new window instead of navigating to the existing one. Also the new window does not have an Lspattached to it and hence I will have to manually open the file again. How do you think I can solve this issue.

Only [showing in split is supported](https://github.com/echasnovski/mini.nvim/blob/8413efde2546be4dec9ea424bc6af346a14dffbe/doc/mini-git.txt#L281-L283). Showing in the same window is not clean enough because returning to the `:Git diff` buffer is an extra hustle as the buffer is not listed. This won't be supported.

LSP server is not attached because opened buffer is not actually a file from disk, but a "snapshot" of it (which is reflected in the name). This [is deliberate](https://github.com/echasnovski/mini.nvim/issues/1126#issuecomment-2275188925).

**Reply by 231tr0n** [reply]
Yup thank you for answering. I was sure this would have already been handled most probably and it is just me not finding it and hence I raised this as a discussion instead of an issue.

**Reply by 231tr0n** [reply]
@echasnovski Also just a small query, like i meant not showing in the same window I mean navigating to an existing buffer or basically opening the buffer newly and normally instead of snapshot and not touching the git diff buffer. Can this be achieved in any way.

**Reply by echasnovski** [reply]
> @echasnovski Also just a small query, like i meant not showing in the same window I mean navigating to an existing buffer or basically opening the buffer newly and normally instead of snapshot and not touching the git diff buffer. Can this be achieved in any way.

This information is [in the next comment](https://github.com/echasnovski/mini.nvim/issues/1126#issuecomment-2276242840) in the issue I linked earlier.

**Reply by 231tr0n** [reply]
Cool let me try it out! Thank you for the reply!!!

**Reply by 231tr0n** [reply]
The solution you suggested works like a charm. Thank you soo much. This fixes all my workflow issues with mini.files and mini-git.

**Reply by 231tr0n** [reply]
@echasnovski just wanted to inform you about my findings since the initial issue was regarding navigating back to the git diff buffer. I used ``tabe`` instead of ``vertical split`` in your given example and I could navigate back to the diff source by closing the buffer opened with ``tabe``. I just wanted to ask if this functionality can be included in mini-git with a different option as part of ``MiniGit.show_diff_source({opts})`` or to that matter as a different function.

**Reply by echasnovski** [reply]
`MiniGit.show_diff_source` [does support showing in separate tab](https://github.com/echasnovski/mini.nvim/blob/f0d6e89515c6b055c20d257da717bd30211532f1/doc/mini-git.txt#L304-L305). It would still be not the file from disk, but special 'mini.git' buffer with the same content and possibly highlighting. No other approach in showing 'mini.git' output (be it `show_at_cursor()`, `:Git`, `show_diff_source()`, and others) apart from already implemented is not planned.
